src/PantosForwarder.sol

```diff
@@ -224,23 +267,34 @@ contract PantosForwarder is IPantosForwarder, Pausable, PantosRBAC {
     function verifyAndForwardTransfer(
         PantosTypes.TransferRequest calldata request,
         bytes memory signature
-    ) external override whenNotPaused onlyPantosHub {
+    ) external override whenNotPaused onlyPantosHub returns (bool, bytes32) {
         // Verify the nonce and signature
         verifyTransfer(request, signature);
         // Mark the nonce as used
         _usedSenderNonces[request.sender][request.nonce] = true;
-        // Transfer the tokens from the sender to the recipient
-        IPantosToken(request.token).pantosTransfer(
-            request.sender,
-            request.recipient,
-            request.amount
-        );
         // Transfer the fee to the service node
         IPantosToken(_pantosToken).pantosTransfer(
             request.sender,
             request.serviceNode,
             request.fee
         );
+        // Transfer the tokens from the sender to the recipient
+        // The transaction is not supposed to be reverted if the token
+        // transfer fails to ensure that the service node gets paid
+        bool succeeded;
+        bytes memory tokenData;
+        (succeeded, tokenData) = request.token.excessivelySafeCall(
```

Comment:
I'm wondering if this is necessary (if this has any benefit on top of a low level call).

As far as I understand, the main reason to use this instead of a low level call is that you can specify how many bytes to copy to memory from the return data. However, IPantosToken.pantosTransfer has no return type, therefore, this would always return 0 bytes as long as we do the call with the IPantosToken.pantosTransfer.selector (which ofc we do). Am I overlooking something?

Reply:
Just because we have it in Solidity code doesn't mean that there is an enforcement on bytecode level. Any token could still simply return some data. If we use here the Solidity call, we don't know if someone is returning data.

Reply:
Hmm, I thought that you can't return anything if you implement a function which has no return type in the signature. How could that be possible?

Reply:
The function signature is calculated without the return part. Only the name and parameter types. As long as a user implements a function with the same name and parameters they can return or not data - as they wish. We do not have any possibility to enforce that a contract does not return data.

Reply:
Any error message of the token contract is also return data.